app/Http/Controllers/CampaignRegistrationController.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Campaign;
+use App\Models\User;
+use App\Models\Character;
+use Illuminate\Support\Facades\Auth;
+
+class CampaignRegistrationController extends Controller
+{
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $data = $request->validate([
+            'character_id' => "required|exists:characters,id",
```

Comment:
Having character ID be required here implies that a user cannot join a campaign as the GM, and so the creator of the campaign must be the GM. Is this a restriction we want to impose?

There are pros and cons for it (such as ensuring there's only ever one GM).
However, I think we should allow for users to join as the GM, if for whatever reason a player is the creator of the campaign.

That would mean character_id here should be 'sometimes'

Reply:
you would also then be expecting a third request parameter being 'is_dm' or something along that line.

Reply:
Right, I wasn't sure if a user could join a campaign as a GM/dm. I can make the adjustment if that is the idea.

Reply:
Yea i think we should leave the option for users to join as GM